plugins/inputs/mongodb/mongodb_server.go

```diff
@@ -257,6 +271,15 @@ func (s *Server) gatherData(acc telegraf.Accumulator, gatherClusterStatus bool,
 		}
 	}
 
+	topStatData := &TopStats{}
+	if gatherTopStat {
+		topStats, err := s.gatherTopStatData()
+		if err != nil {
+			s.Log.Debugf("Unable to gather top stat data: %s", err.Error())
```

Comment:
In the `gatherColStats` path the plugin returns on error. Why do you continue here?

Reply:
My fail, fixed.